core/injector/syncing_node_injector.hpp

```diff
@@ -65,8 +78,10 @@ namespace kagome::injector {
         di::bind<network::OwnPeerInfo>.to(
             [](const auto &injector) { return get_peer_info(injector); }),
 
-        di::bind<consensus::Babe>.template to<consensus::SyncingBabe>(),
-        di::bind<network::BabeObserver>.template to<consensus::SyncingBabe>(),
+        di::bind<consensus::Babe>.to(
+            [](auto const &inj) { return get_syncing_babe(inj); }),
+        di::bind<network::BabeObserver>.to(
+            [](auto const &inj) { return get_syncing_babe(inj); }),
```

Comment:
```suggestion
       di::bind<consensus::Babe>.to<consensus::SyncingBabe>()[di::override],
        di::bind<network::BabeObserver>.to<consensus::SyncingBabe>()[di::override],
```

Reply:
Just checked this works. For block executor, no idea how to make it otherwise

Reply:
Yes, it works